src/components/file-uploader/FileUploader.js

```diff
@@ -0,0 +1,60 @@
+import { useTranslation } from 'react-i18next'
+
+import { Box, Typography, Button, IconButton, ListItem, List } from '@mui/material'
+import CloudUploadIcon from '@mui/icons-material/CloudUpload'
+import CloseIcon from '@mui/icons-material/Close'
+
+import { style } from '~/components/file-uploader/file-uploader.style'
+
+
+const FileUploader = ({ buttonText, upload }) => {
+  const { t } = useTranslation()
+
+  const { dragStart, dragLeave, dragDrop, addFiles, deleteFile, files, isDrag, error } = upload
```

Comment:
I think you should use a hook in this component and emit changes with a callback outside.

Reply:
fixed